registry/handlers/manifests_test.go

```diff
@@ -0,0 +1,174 @@
+package handlers
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"fmt"
+
+	"time"
+
+	"github.com/docker/distribution/configuration"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/api/v2"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/mockdriver"
+)
+
+var errUnexpected = errors.New("unexpected error")
+var testServer = createServer()
+
+func createServer() *httptest.Server {
```

Comment:
@yuwaMSFT2 We use that to test the cross-handler flow, which really ensures that we implement the spec. That is really structured around docs/spec/api.md. You should be able to insert a few cases but I don't recall if we have spots to change upstream errors.